packages/design-system/src/components/web-components/ds-modal-dialog/ds-modal-dialog.stories.tsx

```diff
@@ -120,23 +122,24 @@ const Template = (args) => {
   useEffect(() => {
     const modal = document.querySelector('ds-modal-dialog');
     const toggleButton = document.getElementById('modal-toggle');
+    const form = modal.querySelector('form[method="dialog"]');
 
-    const openModal = () => {
-      modal?.setAttribute('is-open', 'true');
-    };
+    const openModal = () => modal?.setAttribute('is-open', 'true');
+    const closeModal = () => modal?.setAttribute('is-open', 'false');
 
-    modal?.addEventListener('ds-exit', (event) => {
+    const handleExit = (event) => {
       action('ds-exit')(event);
-      modal?.setAttribute('is-open', 'false');
-    });
+      closeModal();
+    };
+
+    modal?.addEventListener('ds-exit', handleExit);
     toggleButton?.addEventListener('click', openModal);
+    form?.addEventListener('submit', closeModal);
 
     return () => {
-      modal?.removeEventListener('ds-exit', (event) => {
-        action('ds-exit')(event);
-        modal?.setAttribute('is-open', 'false');
-      });
+      modal?.removeEventListener('ds-exit', handleExit);
       toggleButton?.removeEventListener('click', openModal);
+      form?.removeEventListener('submit', closeModal);
```

Comment:
Now that the form and buttons are rendered through a `<slot>`, the events are no longer bubbled up to the `<dialog>` in the Shadow Root, which means form submissions do not trigger the `ds-exit` event. I've therefore restructured this example to work with the new paradigm. I've done the same thing to the web component example projects.

Reply:
@jack-ryan-nava-pbc